### ISSUE TITLE
Feature request: MPC Solcast use more than 1 day data

### ISSUE THREAD

**Issue by BDVGitHub** [davidusb-geek/emhass#404]
When at MPC controller the forecast horizon is longer than 1 day. Is it possible to then take into account the Solcast forecast of more than 1 day, since with Solcast you get a 30-minute forecast for up to the next 3 days? Or is this already the case?
![Solcast_Forecast_Gaph](https://github.com/user-attachments/assets/1438874e-f0d7-4fd7-bbe8-04d665c6bf2b)
File with Solcast data in json format [Solcast_Forecasts.json](https://github.com/user-attachments/files/18254305/Solcast_Forecasts.json)

**Comment by GeoDerp** [comment]
Greetings @BDVGitHub ,
yep you should be able to. try setting `historic_days_to_retrieve ` to like `3`.
Alternatively if you [cache solcast](https://emhass.readthedocs.io/en/latest/forecasts.html#caching-solcast) you will automatically have `historic_days_to_retrieve*2` of solcast data cached in EMHASS

**Comment by BDVGitHub** [comment]
Hey @GeoDerp 
In this case, it is not the dates of the days that have passed (historical dates) but the forecast of the days ahead that interests me.
Becuase you are talking about `**historic**_days_to_retrieve` ...

**Comment by GeoDerp** [comment]
I totally get your confusion and thats more my bad for the english of that parameter.
Try passing `historic_days_to_retrieve: 2` as a runtime parameter and see what it comes back with

**Comment by GeoDerp** [comment]
Actually hang on, I reject my statement. Let me look at the code some more.

**Comment by GeoDerp** [comment]
I think your right, and there is more work to be done here. 
@davidusb-geek 
Im thinking of creating a `forecast_days_to_retrieve` param. so we have one for days of past data to retrieve and days of forecast / predicted data.

May look into this, this week, if I have some energy / time after work

**Comment by purcell-lab** [comment]
I agree with this FR it would be good if the EMHASS native solcast would cache multiday future forecasts.

I am able to achieve this outcome by passing data from the solcast sensors as the runtime payload.

```yaml
        "pv_power_forecast": {{
          ([states('sensor.power_photovoltaics')|int(0)] +
          state_attr('sensor.solcast_pv_forecast_forecast_today', 'detailedForecast')|selectattr('period_start','gt',utcnow()) | map(attribute='pv_estimate')|map('multiply',1000)|map('int')|list +
          state_attr('sensor.solcast_pv_forecast_forecast_tomorrow', 'detailedForecast')|selectattr('period_start','gt',utcnow()) | map(attribute='pv_estimate')|map('multiply',1000)|map('int')|list +
          state_attr('sensor.solcast_pv_forecast_forecast_day_3', 'detailedForecast')|selectattr('period_start','gt',utcnow()) | map(attribute='pv_estimate')|map('multiply',1000)|map('int')|list +          
          state_attr('sensor.solcast_pv_forecast_forecast_day_4', 'detailedForecast')|selectattr('period_start','gt',utcnow()) | map(attribute='pv_estimate')|map('multiply',1000)|map('int')|list
          )| tojson
        }},
```

![Image](https://github.com/user-attachments/assets/838cf716-e748-484e-a490-cd9aee46b883)

**Comment by LesIT1** [comment]
Looks like this is addressed by #937, which auto-extends the naive-mpc forecast window to cover a multi-day prediction_horizon when using Solcast, so you can run past 24h without the manual delta_forecast_daily workaround. Might be closeable now.
